doc/MSTP/MSTP.md

```diff
@@ -0,0 +1,1178 @@
+
+
+1. Common and Internal Spanning Tree (CIST): The common and internal spanning tree (CIST) is a single spanning tree that connects all devices in a switched network. It consists of the ISTs in all MST regions and the CST. 
+
+1. MST Instances (MSTIs): MSTP divides the network into multiple regions, each containing several MSTIs. Each MSTI operates independently, allowing for efficient use of network resources and optimized load balancing across different VLANs.
+1.  MST Regions:  An MST region is a group of interconnected bridges that share the same MST configuration, including the MST configuration name, revision number, and VLAN-to-instance mappings
+1. VLAN-to-MSTI Mapping: MSTP maps VLANs to specific MSTIs using a VLAN mapping table. This mapping ensures that traffic within a VLAN follows the corresponding MSTI, optimizing the network path and improving performance. 
+1.  Optimized Bandwidth: By mapping VLANs to specific MSTIs, MSTP ensures that network paths are used efficiently, reducing link blocking and optimizing bandwidth usage
+ 
+#### MSTP Operation 
+
+MST BPDUs: MSTP uses Multiple Spanning Tree Bridge Protocol Data Units (MST BPDUs) to exchange information between switches. These BPDUs contain information about the MSTI and VLAN mappings, ensuring consistent spanning tree calculations across the network. 
+
+MSTP calculates spanning trees on the basis of Multiple Spanning Tree Bridge Protocol Data Units (MST BPDUs).
+
+<div align="center">
+<img src="images/MSTP_BPDU.png" alt="MSTP BPDU">
+<p>MSTP BPDU Format</p>
+</div><br>
+<div align="center">
+<img src="images/MSTI_config_message.png" alt="MSTP Config Message">
+<p>MSTI Configuration Messages</p>
+</div>
+
+*Refer to [RFC IEEE 802.1s-2002](https://standards.ieee.org/ieee/802.1s/1042/) for MSTP BPDU details.* 
+
+
+### Requirements 
+
+
+#### Functional Requirements 
+
+1. Support the creation of Multiple Spanning Tree Instances (MSTIs). 
+1. Support the assignment of one or more VLANs to a specific MSTI within a region. 
+
+1. Support the option to assign a region name and revision number to MSTP regions to achieve unique identification of VLAN to instance mapping across switches. 
+
+1. Support path selection and forwarding behavior in MSTI to optimize network performance within each instance by configuring a distinct root bridge. 
+
+1. Support the configuration of spanning tree parameters such as forward delay, hello timer, hop count. 
+
+1. The Destination Mac Address will be 01:80:C2:00:00:00 for MSTP BPDUs. 
+
+1. Support compatibility with networks employing different spanning tree protocols, such as STP, RSTP. 
+
+1. Support edge port functionality 
+
+1. Support BPDU guard functionality 
+
+1. Support root guard functionality 
+
+1. Support protocol operation on static breakout ports 
+
+1. Support protocol operation on Port-channel interfaces 
```

Comment:
Is the MSTP HLD supports VS image (no ASIC) ?

Reply:
Yes

Reply:
Updated HLD